R/functions-range.R

```diff
@@ -0,0 +1,213 @@
+# TODO: link to the Date type docs
+#' Generate a date range
+#'
+#' If both `start` and `end` are passed as the Date types (not Datetime), and
+#' the `interval` granularity is no finer than `"1d"`, the returned range is
+#' also of type Date. All other permutations return a Datetime.
+#'
+#' @inheritParams rlang::check_dots_empty0
+#' @param start Lower bound of the date range. Something that can be coerced to
+#' a Date or a [Datetime][DataType_Datetime] expression. See examples for details.
+#' @param end Upper bound of the date range. Something that can be coerced to a
+#' Date or a [Datetime][DataType_Datetime] expression. See examples for details.
+#' @param interval Interval of the range periods, specified as a [difftime]
+#' object or using the Polars duration string language. See the `Polars
+#' duration string language` section for details. Must consist of full days.
+#' @param closed Define which sides of the range are closed (inclusive).
+#' One of the following: `"both"` (default), `"left"`, `"right"`, `"none"`.
+#' @return An [Expr][Expr_class] of data type Date or [Datetime][DataType_Datetime]
+#'
+#' @inheritSection polars_duration_string Polars duration string language
+#'
+#' @seealso [`pl$date_ranges()`][pl__date_ranges] to create a simple Series of
+#' data type list(Date) based on column values.
+#'
+#' @examples
+#' # Using Polars duration string to specify the interval:
+#' pl$select(
+#'   date = pl$date_range(as.Date("2022-01-01"), as.Date("2022-03-01"), "1mo")
+#' )
+#'
+#' # Using `difftime` object to specify the interval:
+#' pl$select(
+#'   date = pl$date_range(
+#'     as.Date("1985-01-01"),
+#'     as.Date("1985-01-10"),
+#'     as.difftime(2, units = "days")
+#'   )
+#' )
+pl__date_range <- function(
+    start,
+    end,
```

Comment:
Could you check types of these objects?
Perhaps a dedicated function would be worthwhile.